spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/repository/query/PartTreeDatastoreQuery.java

```diff
@@ -101,15 +109,49 @@ private void validateAndSetFilterParts() {
 
 	@Override
 	public Object execute(Object[] parameters) {
+		Class<?> returnedObjectType = getQueryMethod().getReturnedObjectType();
+		if (isPageQuery()) {
+			List<?> resultEntries = (List) execute(parameters, returnedObjectType, List.class, false);
+			Long totalCount = (Long) execute(parameters, Long.class, null, true);
+
+			ParameterAccessor paramAccessor =
+					new ParametersParameterAccessor(getQueryMethod().getParameters(), parameters);
+
+			return new PageImpl<>(resultEntries, paramAccessor.getPageable(), totalCount);
+		}
+		if (isSliceQuery()) {
+			DatastoreTemplate.LimitedResult result = (DatastoreTemplate.LimitedResult) execute(parameters,
```

Comment:
It looks to me that `sliceQuery` and `LimitedResult` are only used here, and the purpose is only to see if the given limit count is bigger or smaller than the result. 

I think we should be doing that here instead of having a public `sliceQuery` method in DatstoreTemplate and a public inner class `LimitedResult`. The reason is I think those two things don't have a use to our users, and are really implementation things that we currently and foreseeable-y only need here, so why not move them here instead?